internal/netsync/manager.go

```diff
@@ -1793,41 +1821,80 @@ func (m *SyncManager) requestFromPeer(p *peerpkg.Peer, blocks, txs []*chainhash.
 		m.requestedBlocks[*bh] = struct{}{}
 	}
 
-	// Add the vote transactions to the request.
-	for _, vh := range txs {
-		// If we've already requested this transaction, skip it.
-		_, alreadyReqP := state.requestedTxns[*vh]
-		_, alreadyReqB := m.requestedTxns[*vh]
-
-		if alreadyReqP || alreadyReqB {
-			continue
+	addTxsToRequest := func(txs []*chainhash.Hash, txType stake.TxType) error {
+		// Return immediately if txs is nil.
+		if txs == nil {
+			return nil
 		}
 
-		// Ask the transaction memory pool if the transaction is known
-		// to it in any form (main pool or orphan).
-		if m.cfg.TxMemPool.HaveTransaction(vh) {
-			continue
-		}
+		for _, tx := range txs {
+			// If we've already requested this transaction, skip it.
+			_, alreadyReqP := state.requestedTxns[*tx]
+			_, alreadyReqB := m.requestedTxns[*tx]
 
-		// Check if the transaction exists from the point of view of the
-		// end of the main chain.
-		entry, err := m.cfg.Chain.FetchUtxoEntry(vh)
-		if err != nil {
-			return err
-		}
-		if entry != nil {
-			continue
-		}
+			if alreadyReqP || alreadyReqB {
+				continue
+			}
 
-		err = msgResp.AddInvVect(wire.NewInvVect(wire.InvTypeTx, vh))
-		if err != nil {
-			return fmt.Errorf("unexpected error encountered building request "+
-				"for mining state vote %v: %v",
-				vh, err.Error())
+			// Ask the transaction memory pool if the transaction is known
+			// to it in any form (main pool or orphan).
+			if m.cfg.TxMemPool.HaveTransaction(tx) {
+				continue
+			}
+
+			// Check if the transaction exists from the point of view of the end of
+			// the main chain.  Note that this is only a best effort since it is
```

Comment:
s /  ... from the point of view of the end of the main chain. / ... from the point of view of the main chain tip.

Reply:
Updated.